From 761d4c5487570b9c1b37ac4ea13d86fcdb244f65 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Fri, 12 Aug 2011 09:44:45 +0100
Subject: [PATCH] Related: rhbz#730225 avoid segv in ld, this was set to -O0
 around 2000

---
 sd/Library_sdfilt.mk |   12 +-----------
 1 files changed, 1 insertions(+), 11 deletions(-)

diff --git a/sd/source/filter/eppt/makefile.mk b/sd/source/filter/eppt/makefile.mk
index 2a91103..7871436 100644
--- a/sd/source/filter/eppt/makefile.mk
+++ b/sd/source/filter/eppt/makefile.mk
@@ -37,10 +37,6 @@ VISIBILITY_HIDDEN=TRUE
 
 # --- Files --------------------------------------------------------
 
-.IF "$(COM)"=="GCC"
-NOOPTFILES= $(SLO)$/epptso.obj
-.ENDIF
-
 SLOFILES =  $(SLO)$/eppt.obj			\
             $(SLO)$/epptso.obj			\
             $(SLO)$/escherex.obj		\
