From 5e4c9fb9bbf12b40afbefb30e8d71d6e046b5bd0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Fri, 26 Aug 2011 13:05:32 +0100
Subject: [PATCH] Related: fdo#37195 migrationoo3 not registered

---
 postprocess/packcomponents/makefile.mk |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/postprocess/packcomponents/makefile.mk b/postprocess/packcomponents/makefile.mk
index b5f50a0..c23da77 100644
--- a/postprocess/packcomponents/makefile.mk
+++ b/postprocess/packcomponents/makefile.mk
@@ -126,6 +126,7 @@
     lwpfilter \
     mcnttype \
     migrationoo2 \
+    migrationoo3 \
     mtfrenderer \
     msfilter \
     msworksfilter \
