From b90ac7d682fd65f75eff4225d871130c0ae9f185 Mon Sep 17 00:00:00 2001
From: Joshua Cogliati <jjcogliati-r1@yahoo.com>
Date: Fri, 6 Jan 2012 10:48:01 +0000
Subject: [PATCH] smath does not handle accents in MathML

---
 starmath/source/node.cxx |   36 ++++++++++++++++++++++++++++++++----
 1 files changed, 32 insertions(+), 4 deletions(-)

diff --git a/starmath/source/node.cxx b/starmath/source/node.cxx
index 58010e7..d5866cd 100644
--- a/starmath/source/node.cxx
+++ b/starmath/source/node.cxx
@@ -2821,23 +2821,51 @@ void SmAttributNode::CreateTextFromNode(String &rText)
             nLast = aStr.GetChar(0);
             switch (nLast)
             {
-            case 0xAF:
+            case 0xAF: // MACRON
                 APPEND(rText,"overline ");
                 break;
-            case 0x2d9:
+            case 0x2d9: // DOT ABOVE
                 APPEND(rText,"dot ");
                 break;
-            case 0x2dc:
+            case 0x2dc: // SMALL TILDE
                 APPEND(rText,"widetilde ");
                 break;
-            case 0xA8:
+            case 0xA8: // DIAERESIS
                 APPEND(rText,"ddot ");
                 break;
             case 0xE082:
                 break;
             case 0xE09B:
+            case 0x20DB: // COMBINING THREE DOTS ABOVE
                 APPEND(rText,"dddot ");
                 break;
+            case 0x301: // COMBINING ACUTE ACCENT
+                APPEND(rText,"acute ");
+                break;
+            case 0x300: // COMBINING GRAVE ACCENT
+                APPEND(rText,"grave ");
+                break;
+            case 0x30C: // COMBINING CARON
+                APPEND(rText,"check ");
+                break;
+            case 0x306: // COMBINING BREVE
+                APPEND(rText,"breve ");
+                break;
+            case 0x30A: // COMBINING RING ABOVE
+                APPEND(rText,"circle ");
+                break;
+            case 0x20D7: // COMBINING RIGHT ARROW ABOVE
+                APPEND(rText,"vec ");
+                break;
+            case 0x303: // COMBINING TILDE
+                APPEND(rText,"tilde ");
+                break;
+            case 0x302: // COMBINING CIRCUMFLEX ACCENT
+                APPEND(rText,"hat ");
+                break;
+            case 0x304: // COMBINING MACRON
+                APPEND(rText,"bar ");
+                break;
             default:
                 rText.Append(nLast);
                 break;
-- 
1.7.6.5

