diff --git a/solenv/gbuild/platform/linux-S390.mk b/solenv/gbuild/platform/linux-S390.mk
index 410038e..fae49f4 100755
--- a/solenv/gbuild/platform/linux-S390.mk
+++ b/solenv/gbuild/platform/linux-S390.mk
@@ -28,7 +28,7 @@
 #*************************************************************************
 
 #please make generic modifications to unxgcc.mk or linux.mk
-gb_CPUDEFS += -D$(CPUNAME)
+gb_CPUDEFS += -DS390
 gb_Library_DLLPOSTFIX := l3
 gb_COMPILERDEFAULTOPTFLAGS := -O2
 gb_CXXFLAGS += -fsigned-char -fno-omit-frame-pointer
